addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComHomeConfortMessage.java

```diff
@@ -0,0 +1,271 @@
+            unitCode = 0;
+        }
+        signalLevel = (byte) ((data[12] & 0xF0) >> 4);
+    }
+
+    @Override
+    public byte[] decodeMessage() {
+
+        byte[] data = new byte[13];
+
+        data[0] = 0x0C;
+        data[1] = PacketType.HOME_CONFORT.toByte();
+        data[2] = subType.toByte();
+        data[3] = seqNbr;
+        data[4] = (byte) ((deviceId >> 16) & 0xff);
+        data[5] = (byte) ((deviceId >> 8) & 0xff);
+        data[6] = (byte) (deviceId & 0xff);
+        data[7] = (byte) houseCode;
+        data[8] = unitCode;
+        data[9] = command.toByte();
+        data[10] = 0;
+        data[11] = 0;
+        data[12] = (byte) ((signalLevel & 0x0F) << 4);
+
+        return data;
+    }
+
+    @Override
+    public String getDeviceId() {
+        return deviceId + ID_DELIMITER + houseCode + ID_DELIMITER + unitCode;
+    }
+
+    @Override
+    public State convertToState(RFXComValueSelector valueSelector) throws RFXComException {
+
+        State state;
+
+        if (valueSelector.getItemClass() == NumberItem.class) {
+
+            if (valueSelector == RFXComValueSelector.SIGNAL_LEVEL) {
+
+                state = new DecimalType(signalLevel);
+
+            } else {
+                throw new RFXComException("Can't convert " + valueSelector + " to NumberItem");
+            }
+
+        } else if (valueSelector.getItemClass() == SwitchItem.class) {
+
+            if (valueSelector == RFXComValueSelector.COMMAND) {
+
+                switch (command) {
+                    case OFF:
+                    case GROUP_OFF:
+                        state = OnOffType.OFF;
+                        break;
+
+                    case ON:
+                    case GROUP_ON:
+                        state = OnOffType.ON;
+                        break;
+
+                    default:
+                        throw new RFXComException("Can't convert " + command + " to SwitchItem");
+                }
+
+            } else {
+                throw new RFXComException("Can't convert " + valueSelector + " to SwitchItem");
+            }
+
+        } else {
+
+            throw new RFXComException("Can't convert " + valueSelector + " to " + valueSelector.getItemClass());
+
+        }
+
+        return state;
+    }
+
+    @Override
+    public void setSubType(Object subType) throws RFXComException {
+        this.subType = ((SubType) subType);
+    }
+
+    @Override
+    public void setDeviceId(String deviceId) throws RFXComException {
+
+        String[] ids = deviceId.split("\\" + ID_DELIMITER);
+        if (ids.length != 3) {
+            throw new RFXComException("Invalid device id '" + deviceId + "'");
+        }
+
+        this.deviceId = Integer.parseInt(ids[0]);
+        houseCode = ids[1].charAt(0);
+        unitCode = Byte.parseByte(ids[2]);
+    }
+
+    @Override
+    public void convertFromState(RFXComValueSelector valueSelector, Type type) throws RFXComException {
+
+        switch (valueSelector) {
```

Comment:
I didn't want to end up with two different but equivalent ways of doing things. The whole RFXComValueSelector thing should be killed across the board at once.